DigitalLearningSolutions.Data/Services/GroupsService.cs

```diff
@@ -0,0 +1,234 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Transactions;
+    using DigitalLearningSolutions.Data.DataServices;
+    using DigitalLearningSolutions.Data.Helpers;
+    using DigitalLearningSolutions.Data.Models.DelegateGroups;
+    using DigitalLearningSolutions.Data.Models.Email;
+    using DigitalLearningSolutions.Data.Models.User;
+    using MimeKit;
+
+    public interface IGroupsService
+    {
+        void SynchroniseUserChangesWithGroups(
+            DelegateUser oldDelegateDetails,
+            AccountDetailsData newDelegateDetails,
+            CentreAnswersData newCentreAnswers,
+            string baseUrl
+        );
+
+        IEnumerable<Group> GetSynchronisedGroupsForCentre(int centreId);
+
+        void EnrolDelegateOnGroupCourses(

```

Comment:
Are these 2 only public for unit testing or are they going to be used in the future?

Reply:
EnrolDelegateOnGroupCourses should end up being used on https://softwiretech.atlassian.net/browse/HEEDLS-514 (which is mentions in the ticket for this), but it is also useful for unit testing. GetSynchronisedGroupsForCentre I think will only be used here, but is unit tested, though admittedly its a very simple method that could get incorporated into the logic if we'd prefer?

Reply:
I think remove GetSynchronisedGroupsForCentre from the interface then and make it private. Makes it easier to bring back if we do decide we need it again outside this service.